dpnp/dpnp_iface_histograms.py

```diff
@@ -337,3 +340,88 @@ def histogram(a, bins=10, range=None, density=None, weights=None):
         return n / db / n.sum(), bin_edges
 
     return n, bin_edges
+
+
+def histogram_bin_edges(a, bins=10, range=None, weights=None):
+    r"""
+    Function to calculate only the edges of the bins used by the
+    :obj:`dpnp.histogram` function.
+
+    For full documentation refer to :obj:`numpy.histogram_bin_edges`.
+
+    Parameters
+    ----------
+    a : {dpnp.ndarray, usm_ndarray}
+        Input data. The histogram is computed over the flattened array.
+    bins : {int, dpnp.ndarray, usm_ndarray, sequence of scalars}, optional
+        If `bins` is an int, it defines the number of equal-width bins in the
+        given range.
+        If `bins` is a sequence, it defines the bin edges, including the
+        rightmost edge, allowing for non-uniform bin widths.
+        Default: ``10``.
+    range : {None, 2-tuple of float}, optional
+        The lower and upper range of the bins. If not provided, range is simply
+        ``(a.min(), a.max())``. Values outside the range are ignored. The first
+        element of the range must be less than or equal to the second. `range`
+        affects the automatic bin computation as well. While bin width is
+        computed to be optimal based on the actual data within `range`, the bin
+        count will fill the entire range including portions containing no data.
+        Default: ``None``.
+    weights : {None, dpnp.ndarray, usm_ndarray}, optional
+        An array of weights, of the same shape as `a`. Each value in `a` only
+        contributes its associated weight towards the bin count (instead of 1).
+        This is currently not used by any of the bin estimators, but may be in
+        the future.
+        Default: ``None``.
+
+    Returns
+    -------
+    bin_edges : {dpnp.ndarray of floating data type}
+        The edges to pass into :obj:`dpnp.histogram`.
+
+    See Also
+    --------
+    :obj:`dpnp.histogram` : Compute the histogram of a data set.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> arr = np.array([0, 0, 0, 1, 2, 3, 3, 4, 5])
+    >>> np.histogram_bin_edges(arr, bins=2)
+    array([0. , 2.5, 5. ])
+
+    For consistency with histogram, an array of pre-computed bins is
+    passed through unmodified:
+
+    >>> np.histogram_bin_edges(arr, [1, 2])
+    array([1, 2])
+
+    This function allows one set of bins to be computed, and reused across
+    multiple histograms:
+
+    >>> shared_bins = np.histogram_bin_edges(arr, bins=5)
+    >>> shared_bins
+    array([0., 1., 2., 3., 4., 5.])
+
+    >>> gid = np.array([0, 1, 1, 0, 1, 1, 0, 1, 1], sycl_queue=arr.sycl_queue)
```

Comment:
I am not sure you should pass `sycl_queue` here.
According to the examples above `gid` should be allocated on the same queue as `arr`

Reply:
It's because the default queue is used, in case when you consequently run the code from the example.
But in general, `arr` might be allocating on another queue and when we must secure `gid` is allocating on the same queue to comply with compute follows data.
By passing explicitly `sycl_queue=arr.sycl_queue` I wanted to point on that one more time.

Reply:
Maybe it would be better to leave a comment for this line?
What do you think?

Reply:
Why do you think it would be better?

Reply:
Сause in another examples, for example, in dpnp.searsorted([297 line](https://github.com/IntelPython/dpnp/blob/master/dpnp/dpnp_iface_searching.py#L297)), we do not pass sycl_queue for `v`

Reply:
Ok, removed passing `sycl_queue` keyword from the example.
Btw, I still don't feel that having it in this example was confusing. It's just one of the possible way to allocate an array. But since it doesn't relate to reviewing functionality I agreed to align with examples of other functionality.

We can discuss if we would like to pass `sycl_queue` in the examples separately.